subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -56,6 +56,15 @@
 
 #define DISCOVER_PREFACE	"</.well-known/core>;ct=40"
 
+#define RESOURCE_TYPE		";rt=\"oma.lwm2m\""
```

Comment:
I'm not sure we want to implement the alternate path tag when we aren't using an alternate path.  It's an optional tag used when the \</\> root resource location is different such as \</lwm2m\>;rt=\"oma.lwm2m\"

I suppose we could add a Kconfig option to allow the user the option of setting the alternate path.  An example would be config LWM2M_RD_CLIENT_ALTERNATE_PATH as a string defaulting to not set.   And then if that value is set we need to insert it everywhere in the registration string as well as add the ;rt="oma.lwm2m" portion.

Did you need to use alternate path for some reason?